federation/pkg/crud/secret.go

```diff
@@ -0,0 +1,93 @@
+/*
```

Comment:
Didn't this code exist somewhere else already? I thought I reviewed this previously. Is there something you can base this off of?